.github/workflows/validate.yml

```diff
@@ -424,7 +441,7 @@ jobs:
       - name: Install cabal-plan
         run: |
           cd $(mktemp -d)
-          cabal install cabal-plan --constraint='cabal-plan +exe'
+          cabal install cabal-plan --allow-newer="base,ghc-prim,template-haskell" --constraint='cabal-plan +exe'
```

Comment:
apply this unconditionally have a (small?) risk of breaking cabal-plan install for other matrix combinations

Reply:
I think it's not worth bothering about it. And right now it's just `base`.

Moreover, I have a hunch that cabal-plan is not needed at all these days in our setup. I plan to check it and remove if that's the case.